server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -960,6 +960,23 @@ public boolean doesConsumerExist(String uuid) {
         return result != 0;
     }
 
+    public List<String> getNonExistingConsumer(ArrayList<String> listOfUids) {
+        DetachedCriteria criteria = this.createSecureDetachedCriteria()
+            .add(Restrictions.in("uuid", listOfUids));
+
+        CandlepinQuery<Consumer> consumersFound = this.cpQueryFactory.
+            <Consumer>buildQuery(this.currentSession(), criteria);
+
+        List<String> collectFoundIds = new ArrayList<>();
+
+        for (Consumer co : consumersFound) {
```

Comment:
This loop and the collection is unnecessary. If you're going to keep the general behavior of modifying the input list (which I would argue against), iterate through it and remove the elements you get back from the query.

Reply:
Yes. Fixing it. Removing it from same Input Collection to avoid creating a copy of it.